### PR TITLE
[dbsp] Avoid scheduling cycles caused by delayed integrals.

### PR DESCRIPTION
**Pull request by ryzhyk**
Solves #69.

The actual fix is simple, but it required fixing a related issue, which involved a bigger change. See commit messages for details.

### CODE REVIEW

File: crates/dbsp/src/trace/ord/vec/indexed_wset_batch.rs [hunk 1]
```diff
@@ -532,10 +528,10 @@ where
         _dst_hint: Option<BatchLocation>,
     ) -> Self {
         Self {
-            lower1: batch1.layer.lower_bound(),
-            upper1: batch1.layer.lower_bound() + batch1.layer.keys(),
-            lower2: batch2.layer.lower_bound(),
-            upper2: batch2.layer.lower_bound() + batch2.layer.keys(),
+            lower1: 0,
```
Comment (mihaibudiu) [inline crates/dbsp/src/trace/ord/vec/indexed_wset_batch.rs:531]: so these are still used?

**Comment by blp** [comment]
Before I properly review this, I'm just going to say how delighted I am that truncate_keys_below is going away!

**Review by mihaibudiu — APPROVED** [review]
Maybe you can remove some more stuff?

File: crates/dbsp/src/trace/ord/vec/indexed_wset_batch.rs (hunk 1 above)
Comment (blp) [reply crates/dbsp/src/trace/ord/vec/indexed_wset_batch.rs:531]: Still needed, the merger advances the lower pointers until they reach the upper pointers.

**Review by blp — APPROVED** [review]
This is excellent. Thank you.

File: crates/dbsp/src/trace/ord/vec/indexed_wset_batch.rs (hunk 1 above)
Comment (ryzhyk) [reply crates/dbsp/src/trace/ord/vec/indexed_wset_batch.rs:531]: yeah, these are used to track the position of a cursor.

**Pull request merged**